### PR TITLE
[Feature] Implements Graphql API

### PR DESCRIPTION
**Pull request by apoclyps**
### What's Changed

Implements a GraphQL API on top of the existing models/implementation for `comic_books`. This change is backwards compatible with the existing view logic as it extends the models directly.

* Adds queries and mutations for creating a comic book
* Adds snapshot tests for testing queries and mutations easily

Notable, this change will allow others (including myself) to build upon the work produced on the server to create multiple clients (Web, Mobile, etc) using any GraphQL client. In addition, the following video/blog provides an excellent overview of the benefits of [GraphQL vs Rest](https://www.howtographql.com/basics/1-graphql-is-the-better-rest/).

### Technical Description
[GraphQL](https://graphql.org/) is a query language for APIs and a runtime for fulfilling those queries with your existing data. GraphQL provides a complete and understandable description of the data in your API, gives clients the power to ask for exactly what they need and nothing more, makes it easier to evolve APIs over time, and enables powerful developer tools.

The following example shows `graphiql`, a web-based UI that runs within the Django application to query against the service.

image

### Dependencies
Depends on #8 in order to run the test suite correctly.

### CODE REVIEW

**Comment by ajurna** [comment]
why create directory? do you plan on adding a method of adding comics via an api?
usually directories are scanned when queried. so there doesn't need to be an update/create methods as they should update dynamically.

**Comment by apoclyps** [comment]
@ajurna yes, that's almost exactly what I would like to do. The PR is the prerequisite to building out a much richer GraphQL API to support a React client (web, desktop, mobile). This PR lay's the initial groundwork for introducing a non-breaking API, tests, and example queries. A feature I would like to implement is the ability to upload comics from another client to the server and view those on any device (this is in part why I've introduced the above).  Happy to make changes to the above (if needed).

**Comment by ajurna** [comment]
I manage my comic library with mylar. Would you be looking at implementing similar functionality? Currently everything is very heavily dependent on the filesystem and doesnt have any meta information to identify files.

File: comic/api/mutations.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        id = graphene.Int(required=True)
+        input = DirectoryInput(required=True)
+
+    ok = graphene.Boolean()
+    directory = graphene.Field(DirectoryType)
+
+    @staticmethod
+    def mutate(root, info, id, input=None):
+        ok = False
+        directory = Directory.objects.get(pk=id)
```
Comment (ajurna) [inline comic/api/mutations.py]: This will raise an exception. Needs a filter and exists to work this way

File: comic/api/mutations.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    class Arguments:
+        input = DirectoryInput(required=True)
+
+    ok = graphene.Boolean()
+    directory = graphene.Field(DirectoryType)
+
+    @staticmethod
+    def mutate(root, info, input=None):
+        ok = True
+        directory = Directory(name=input.name)
```
Comment (ajurna) [inline comic/api/mutations.py]: There is no reference to a parent directory here. You will only be able to make dirs at the root level

File: comic/api/mutations.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    @staticmethod
+    def mutate(root, info, input=None):
+        ok = True
+        directory = None
+
+        if input.directory:
+            directory = Directory.objects.get(pk=input.directory[0].id)
+
+        comic_book = ComicBook(file_name=input.file_name, directory=directory)
```
Comment (ajurna) [inline comic/api/mutations.py]: This should be done by process_comic_book static method. This makes sure all the pages are loaded correctly

**Comment by ajurna** [comment]
This api is missing any reference to the comicpage model. This would be needed to get filenames in each archive

File: comic/api/mutations.py (hunk 3 above)
Comment (ajurna) [reply comic/api/mutations.py]: the way the current system works. you may be better to upload the file first and then trigger the process comic book. it's either that or there will be a need to refactor the way comics are added and processed.

**Pull request closed without merging**